### ISSUE TITLE
fix kv pair overflow bug

### ISSUE THREAD

**Issue by wrgr** [openconnectome/ndio#74]
Overflow/cache bug persists as documented in the attached.  Moving slack discussion file here. 

@j6k4m8 - this does indeed stop a lot of the claims with ndio (anything involving kv pairs, but lots of other ones to work on in the meantime.  :)

![pasted image at 2016_02_15 04_01 pm](https://cloud.githubusercontent.com/assets/884780/13063412/cc7efc5a-d415-11e5-8c00-92c4d739ae35.png)

**Comment by wrgr** [comment]
Ewww....

So there's something funky with kvpairs. 

This works.  Marvel at this awesome python code:

```

import numpy as np
import ndio.remote.neurodata as neurodata
import ndio.ramon as ramon
import time
import ndio

start = time.time()

token = 'kasthuri2015_ramon_v2'
channel = 'neurons_meta'
res = 3

nd = neurodata()

id_segment = nd.get_ramon_ids(token, channel, ramon_type=ramon.RAMONSegment)
id_neuron = nd.get_ramon_ids(token, channel, ramon_type=ramon.RAMONNeuron)
print 'Number of RAMON segments is: ' + str(len(id_segment))
print 'Number of RAMON neruons is: ' + str(len(id_neuron))

import copy
# Get all segments
token = 'kasthuri2015_ramon_v2'
channel = 'neurons_meta'
segAll = []
for x in range(5):
    s = nd.get_ramon_metadata(token,channel,[id_segment[x]])
    segAll.append(copy.deepcopy(s))
```

I assert this should work, but it doesn't.  Would also be nice not to pass in arrays `[ ]`.  By doesn't work, I mean that all parts of the RAMONObject work, except the key value pairs, which get overwritten by the kv pairs in the last object written, as though it were part of a mutable list.   Help!  

```
import numpy as np
import ndio.remote.neurodata as neurodata
import ndio.ramon as ramon
import time
import ndio

start = time.time()

token = 'kasthuri2015_ramon_v2'
channel = 'neurons_meta'
res = 3

nd = neurodata()

id_segment = nd.get_ramon_ids(token, channel, ramon_type=ramon.RAMONSegment)
id_neuron = nd.get_ramon_ids(token, channel, ramon_type=ramon.RAMONNeuron)
print 'Number of RAMON segments is: ' + str(len(id_segment))
print 'Number of RAMON neruons is: ' + str(len(id_neuron))

# Get all segments
token = 'kasthuri2015_ramon_v2'
channel = 'neurons_meta'
segAll = []
for x in range(5):
    segAll.append(nd.get_ramon_metadata(token,channel, [id_segment[x]]))
```

**Comment by wrgr** [comment]
Sounds like refactoring the way we get hdf5 data might help here per lab discussion this afternoon?  Maybe we just avoid the situation?

**Comment by j6k4m8** [comment]
Agreed re refactoring... Definitely will need some more attention and discussion this week, I think...

**Comment by wrgr** [comment]
still broken?  this is holding on the json interface, right?  And that's now ready?  If so, let's knock this one out.

**Comment by j6k4m8** [comment]
Fixed in #147
